src/main/scala/org/dbpedia/spotlight/wikistats/ComputeStats.scala

```diff
@@ -234,22 +232,15 @@ class ComputeStats(lang: String) (implicit val sc: SparkContext,implicit val sql
       val snowballStemmer = new SnowballStemmer(stemmerString)
       val locale = new Locale(language)
       val list = new LanguageIndependentStringTokenizer(locale,snowballStemmer)
-
+      val dbpediaEncode = new DBpediaUriEncode(language)
       val stemStopWords = collection.mutable.Set[String]()
       SpotlightUtils.createStopWordsSet(stopWordLoc).foreach(word =>
         list.tokenize(word).foreach(stemWord => stemStopWords += stemWord))
 
-      part.flatMap(row => list.tokenize(row._2).filter(!stemStopWords.contains(_)).toList.map(token => (row._1,token)))
-    }).toDF("uri","token")
-      .groupBy("uri","token")
-      .count
-      .rdd
-      .mapPartitions{rows =>
-    {
-      val dbpediaEncode = new DBpediaUriEncode(language)
-      rows.map(row =>
-        (dbpediaEncode.wikiUriEncode(row.getString(0)),(row.getString(1),row.getLong(2)).toString()))}}
-      .reduceByKey(_ + _)
+      part.map(row => (dbpediaEncode.wikiUriEncode(row._1),SpotlightUtils.countTokens(list.tokenize(row._2).filter(!stemStopWords.contains(_)).toList)))
```

Comment:
Yes, That is to avoid extra shuffle and counting the tokens locally. 
I think this is efficient, we will come to know when i run the whole dataset. ;)